joomla/joomla-cms
# [4.0][RFC] Drop minified static files from the repo

Pull Request for Issue # .

### Summary of Changes
- All minified css and js files were removed from the repo
- JHTML (or better HTMLHelper) was adjusted to pick the right files depending on debug state and irrelative to the given extension (eg .min.js or .js)

### Why
- This will make contributors life easier as they won't have to mess with node and try to figure out how to minify js.
- It's kinda unsafe to accept minified code...

### Possible problems
- The repo remains installable but doesn't contain the production preferred minified files, so someone downloading it from GitHub will have to run build in order to get to production mode.
- Although the idea here is to help contributors by offering an easer path for involvement and there is some security issue by accepting minified code, we do increase our dependency on tools and maybe this requires some more testing. Probably the release team can shed some light here.

### WIP
As I opened the PR for comments there are parts that are missing here:
- For SASS we need to see if we can trigger the minifier per request, so things could be tested.
- The build script needs to run the node uglifier then pack the packages and then remove the minified files again. This code is not included in the PR (it makes no sense to spent time if this won't be approved)

### Testing Instructions
Apply patch, or better download the branch from issue tracker and install. Check debug options and that all css and js files are still loaded correctly


### Expected result



### Actual result



### Documentation Changes Required
Yes but mainly this will affect the maintainers (for the rest contributors is just removing the requirement for compressing js)

Calling @mbabker @wilsonge @rdeutz @dneukirchen @yvesh
